### ISSUE TITLE
'/calibre' URL Hard-Wired Somewhere?

### ISSUE THREAD

**Issue by madranet** [janeczku/calibre-web#1241]
I've run into an odd problem with setting up Calibre-Web on HTTPS on a Ubuntu server.  

Everything works fine, if I use the default of proxying `mydomain.com/calibre` to Calibre-Web.  However, if I change my proxy settings to use a different URL for my library -- I wanted `mydomain.com/bookshelf` -- then I get a 404 served up by Calibre-Web:

![SS 2020-03-04 at 11 16 26 (2)](https://user-images.githubusercontent.com/669658/75874954-b1916080-5e0a-11ea-8307-9c8b1d187176.png)

Clicking on the 'Return to Home' link on that page then gives me another 404 --this time served up by my webserver itself and, as you can see from the URL bar, the 'Return to Home' link has redirected to `/calibre` instead of `/bookshelf` as I have in my proxy settings.  

![SS 2020-03-04 at 11 16 59 (2)](https://user-images.githubusercontent.com/669658/75875125-046b1800-5e0b-11ea-8219-efb9db0861e6.png)

So it seems the `/calibre` URL is hard-wired in somewhere.  Is this normal practice for a proxy redirect? I'm sure in other setups, I've been able to customise the URL I'm  proxying from, though I could be wrong on that --it's been a while since I last set one up!

Just for the record, I'm using Caddy Server and my proxy config is as follows. But, as I said, everything works perfectly if I change the proxy line to `proxy /calibre 127.0.0.1:32323`

```caddy
#redirect www
www.domain.com {
    redir https://domain.com{uri}
}

domain.com {

        <snip>

    #Calibre-Web
    proxy /bookshelf 127.0.0.1:32323 {
    header_upstream X-Script-Name /calibre
    }
}
```

**Comment by darkalchemy** [comment]
I use this without a sub folder: 

```
server {
    listen 80;
    server_name read.mydomain.com;

    access_log off;
    error_log off;

    client_max_body_size 20M;
    location / {
        proxy_bind              $server_addr;
        proxy_pass              http://127.0.0.1:8083;
        proxy_set_header        Host            $http_host;
        proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header        X-Scheme        $scheme;
    }
}
```

and it works without issue.

**Comment by madranet** [comment]
Thanks. 

But I  was trying to avoid having to use a subdomain, as I find them messy to type. 

There's also the issue that if you're sharing your library URL with other less computer-savvy people and you say _"go to 'books.mydomain.com'"_, 99% of them will type _"www.books.mydomain.com"_ into their browser and then tell you it didn't work.

**Comment by OzzieIsaacs** [comment]
I absolutly sure it's not hardcoded, as I'm using it personally with different names and it works without any problem (apache config) and also with https. If we get an error from calibre-web, could you please check which URL calibre-web tries to reach (in developer console of your browser).

**Comment by OzzieIsaacs** [comment]
Looks like you are using the darf theme, I'm using the light/standard theme, could you please change it to this theme and check again. Just to avoid that in nthe dark theme something is wrong

**Comment by madranet** [comment]
I still got the same outcome when switching to the standard theme.  

I also found that, although I still got a 404, I could navigate to the homepage indirectly by typing  the URL to the admin section `/bookshelf/admin/view` directly into the browser's address bar. But all the images were broken and [again] a links on the navbar  wanted to go to `/calibre` rather than `/bookshelf`

![SS 2020-03-04 at 16 11 00](https://user-images.githubusercontent.com/669658/75901959-372b0580-5e37-11ea-98f3-65b5d1cbbc78.png)

It's no big deal. I'll just use the default `/calibre` address. I just found it puzzling that it looked like there was a hard-wired `/calibre` path in there somewhere. But, if you're sure there's not, it's likely some glitch in my Caddy proxy settings. It can be a bit temperamental at times.
